### PR TITLE
Adds a post on instruction-tuning Stable Diffusion

### PR DESCRIPTION
**Pull request by sayakpaul**
Internal document: https://www.notion.so/huggingface2/Instruction-tuning-Stable-Diffusion-with-InstructPix2Pix-8062a8ef65164a609ed56da5f448ddb3

Cc: @pcuenca @kashif @alaradirik 

Have set the date to 23rd May, 2023.

### CODE REVIEW

**Comment by sayakpaul** [comment]
Have reviewed on codespaces. Looking good.

**Review by osanseviero — COMMENTED** [review]
Very cool!

File: instruction-tuning-sd.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
+---
+title: "Instruction-tuning Stable Diffusion with InstructPix2Pix" 
+thumbnail: assets/instruction_tuning_sd/thumbnail.png
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply instruction-tuning-sd.md:3]: @Reeray behind the wizardry 🪄

File: instruction-tuning-sd.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+| ![schematic](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/instruction-tuning-sd/schematic.png) | 
+|:--:|
+| **Figure 1**: We explore the instruction-tuning capabilities of Stable Diffusion. In this figure, we prompt an instruction-tuned Stable Diffusion system with prompts involving different transformations and input images.  The tuned system seems to be able to learn these transformations stated in the input prompts. Figure best viewed in color and zoomed in. |
+
+This idea of teaching Stable Diffusion to follow user instructions to perform **edits** on input images was introduced in [InstructPix2Pix: Learning to Follow Image Editing Instructions](https://arxiv.org/abs/2211.09800). We discuss how to extend the InstructPix2Pix training strategy to follow more specific instructions related to tasks in image translation (such as cartoonization) and low-level image processing (such as image deraining). We cover:
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply instruction-tuning-sd.md:19]: Let me do that!

File: instruction-tuning-sd.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+| ![flan_dataset_overview](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/instruction-tuning-sd/flan_dataset_overview.png) | 
+|:--:|
+| **Figure 3**: FLAN V2 training and test task mixtures (figure taken from the FLAN V2 paper). |
+
+Our motivation behind this work comes partly from the FLAN line of work and partly from InstructPix2Pix. We wanted to explore if it’s possible to prompt Stable Diffusion with specific instructions and input images to process them as per our needs. 
+
+The [pre-trained InstructPix2Pix models](https://huggingface.co/timbrooks/instruct-pix2pix) are good at following general instructions, but they may fall short of following instructions involving specific transformations:
+
+| ![cartoonization_results](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/instruction-tuning-sd/cartoonization_results.png) |
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply instruction-tuning-sd.md:60]: I know :( 

Actually the original image was 26 MB!

I did a two-pass compression and 13 MB is what I could get. Any further suggestions? (I have used ImageOptim and TinyPNG already)

[the remark this answers is not in the document]
Comment (sayakpaul) [reply instruction-tuning-sd.md]: New problems require new solutions haha!

Here's what I did get the current (apparently already optimized PNG) to get to under 2 MBs:

* Downloaded the image:

```bash
wget https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/instruction-tuning-sd/cartoonization_results.png
```

* Loaded the PNG with PIL:

```python
import PIL 

image = PIL.Image.open("cartoonization_results.png")
```

* Then serialized it as a JPEG:

```python
image.convert('RGB').save("cartoonization_results.jpeg")
```

* Downloaded the JPEG variant and further optimized it with TinyPNG.

File: instruction-tuning-sd.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sayakpaul) [reply instruction-tuning-sd.md]: Done.

**Pull request merged**